apis/v1alpha1/validation.go

```diff
@@ -57,3 +62,39 @@ func ValidateIAMRequest(r *IAMRequest) (retErr error) {
 	}
 	return
 }
+
+// ValidateCLIRequest checks if the CLIRequest is valid.
+func ValidateCLIRequest(r *CLIRequest) (retErr error) {
+	// Set default CLI
+	if r.CLI == "" {
+		r.CLI = defaultCLI
+	}
+	// TODO (#49): support other CLIs.
+	if r.CLI != defaultCLI {
+		retErr = errors.Join(retErr, fmt.Errorf("CLI %q is not supported", r.CLI))
+	}
+
+	// Check if the do commands are valid.
+	for _, c := range r.Do {
+		if err := checkCommand(c); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("command %q is not valid: %w", c, err))
+		}
+	}
+
+	// Check if the cleanup commands are valid.
+	for _, c := range r.Cleanup {
+		if err := checkCommand(c); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("command %q is not valid: %w", c, err))
+		}
+	}
+	return retErr
+}
+
+func checkCommand(c string) error {
+	for _, o := range invalidCommandOperators {
+		if strings.Contains(c, o) {
+			return fmt.Errorf(`command contains invalid operators in %q`, invalidCommandOperators)
+		}
+	}
+	return nil
+}
```

Comment:
This is pretty inefficient (searches the string 4 times) and probably not complete. I would suggest:

```go
	invalidCommandOperators = map[string]struct{}{
    "&": {}, 
    "|": {}, 
    ">": {}, 
    ">>": {},
}
```

and then iterate over each character in the string and see if it's in the set of forbidden characters:

```go
for _, ch := range c {
  if _, ok := invalidCommandOperations[ch]; ok {
    return fmt.Errorf("invalid command operator %q", ch)
  }
}
```

Reply:
Got it, updating!